### PR TITLE
Tag support for crds and update from runtime and codegen v0.6.0

### PR DESCRIPTION
**Pull request by ryansteakley**
Enables tag supports for crds and adds them to the e2e testing.
Additionally updates to codegen and runtime v0.6.0
Files to review
- `generator.yaml`
- `test/e2e/resources/`
-  `test/e2e/tests/`

### CODE REVIEW

**Comment by akartsky** [comment]
Too many file changes to review 
Can you mention which files did you modify to get the Tags to work ?

Integration test changes look fine

File: test/e2e/__init__.py [hunk 1]
```diff
@@ -156,3 +157,21 @@ def assert_endpoint_status_in_sync(endpoint_name, reference, expected_status):
...
+    response_tags = response["Tags"]
+
+    while "NextToken" in response:
+        response = sagemaker_client().list_tags(
+            ResourceArn=resource_arn, NextToken=response["NextToken"]
+        )
+        response_tags.extend(response["Tags"])
+
+    # SageMaker returns tags with Key,Value while ACK returns key,value
+    for i in range(len(response_tags)):
```
Comment (akartsky) [inline test/e2e/__init__.py:173]: you can just do 
`response_tags = [{"key": d["Key"], "value": d["Value"]} for d in response_tags]`

File: generator.yaml [hunk 2]
```diff
@@ -22,6 +22,14 @@ operations:
   StopHyperParameterTuningJob:
     operation_type: Delete
     resource_name: HyperParameterTuningJob
+  DescribeModelQualityJobDefinition:
```
Comment (akartsky) [inline generator.yaml:25]: these changes seem unrelated ?

Comment (ryansteakley) [reply generator.yaml:25]: These were required changes from v.0.4.0 since those changes were never merged in these are still required changes in v.0.6.0

File: test/e2e/__init__.py (hunk 1 above)
Comment (ryansteakley) [reply test/e2e/__init__.py:173]: sure

**Review by akartsky — APPROVED** [review]

File: generator.yaml [hunk 3]
```diff
@@ -609,4 +617,4 @@ ignore:
     operations:
       - UpdateTrainingJob
     shape_names:
```
Comment (surajkota) [inline generator.yaml:619]: delete whole shape_names instead of commenting

File: pkg/resource/hyper_parameter_tuning_job/delta.go [hunk 4]
```diff
@@ -110,6 +110,7 @@ func newResourceDelta(
 			delta.Add("Spec.HyperParameterTuningJobName", a.ko.Spec.HyperParameterTuningJobName, b.ko.Spec.HyperParameterTuningJobName)
 		}
 	}
+
```
Comment (surajkota) [inline pkg/resource/hyper_parameter_tuning_job/delta.go:113]: was this automatically added ?

File: templates/training_job/sdk_delete_pre_build_request.go.tpl [hunk 5]
```diff
@@ -2,5 +2,5 @@
 	// resource Unmanaged
 	latestStatus := r.ko.Status.TrainingJobStatus
 	if latestStatus != nil && *latestStatus != svcsdk.TrainingJobStatusInProgress {
-		return nil
```
Comment (surajkota) [inline templates/training_job/sdk_delete_pre_build_request.go.tpl:5]: (not in scope of this PR) looks like a lot of common code can be refactored into common code

File: test/e2e/common/fixtures.py [hunk 6]
```diff
@@ -55,6 +55,7 @@ def xgboost_churn_endpoint(sagemaker_client):
         logging.debug(
             f"ARN for this resource is None, resource status is: {model_resource['status']}"
         )
+
```
Comment (surajkota) [inline test/e2e/common/fixtures.py:58]: ?

File: test/e2e/resources/xgboost_model_with_model_location.yaml [hunk 7]
```diff
@@ -13,4 +13,11 @@ spec:
       environment:
         my_var: my_value
         my_var2: my_value2
-  executionRoleARN: $SAGEMAKER_EXECUTION_ROLE_ARN
\ No newline at end of file
+  executionRoleARN: $SAGEMAKER_EXECUTION_ROLE_ARN
+  tags:
+    - key: algorithm
```
Comment (surajkota) [inline test/e2e/resources/xgboost_model_with_model_location.yaml:18]: dont need here since its not part of model test

File: test/e2e/__init__.py [hunk 8]
```diff
@@ -156,3 +157,18 @@ def assert_endpoint_status_in_sync(endpoint_name, reference, expected_status):
...
+
+    while "NextToken" in response:
+        response = sagemaker_client().list_tags(
+            ResourceArn=resource_arn, NextToken=response["NextToken"]
+        )
+        response_tags.extend(response["Tags"])
+
+    # SageMaker returns tags with Key,Value while ACK returns key,value
+    response_tags = [{"key": d["Key"], "value": d["Value"]} for d in response_tags]
+    TestCase().assertCountEqual(response_tags, resource_tags)
```
Comment (surajkota) [inline test/e2e/__init__.py:174]: this is nice, but can you check if it's possible to compare the values as well? maybe convert to JSON?

or just add 2 tags instead of 3

Comment (ryansteakley) [reply test/e2e/__init__.py:174]: assertCountEqual() checks the length of the lists as well as the values inside them aswell. If dict keys or values are not the same it will throw an assertion error

ex:
AssertionError: Element counts were not equal:
First has 1, Second has 0:  {'key': 'hello', 'value': 'a'}
First has 0, Second has 1:  {'key': 'hello', 'value': 'b'}`

File: pkg/resource/hyper_parameter_tuning_job/delta.go (hunk 4 above)
Comment (ryansteakley) [reply pkg/resource/hyper_parameter_tuning_job/delta.go:113]: Appears so I've rebuilt the code several times and that newline appears

File: templates/common/sdk_delete_post_request.go.tpl [hunk 9]
```diff
@@ -2,8 +2,8 @@
 if err == nil {
         if _, err := rm.sdkFind(ctx, r); err != ackerr.NotFound {
                 if err != nil {
-                        return err
```
Comment (surajkota) [inline templates/common/sdk_delete_post_request.go.tpl:5]: nit: maybe run go fmt on both the template files and lets stick with that. indentation keeps changing between PRs

**Review by surajkota — CHANGES_REQUESTED** [review]

File: templates/common/sdk_delete_post_request.go.tpl (hunk 9 above)
Comment (ryansteakley) [reply templates/common/sdk_delete_post_request.go.tpl:5]: auto formatted :)

**Review by surajkota — APPROVED** [review]
/approve
/lgtm
